### PR TITLE
optimize: remove igonreQueryOrder option

### PR DESCRIPTION
**Pull request by li-jin-gou**
#### What type of PR is this?

optimize

#### What this PR does / why we need it (en: English/zh: Chinese):

en: remove igonreQueryOrder option

### CODE REVIEW

File: cache.go [hunk 1]
```diff
@@ -190,32 +190,51 @@ func newCache(
...
-		cacheStrategy = func(ctx context.Context, c *app.RequestContext) (bool, Strategy) {
-			newUri, err := getRequestUriIgnoreQueryOrder(c.Request.URI().String())
-			if err != nil {
-				hlog.CtxErrorf(ctx, getRequestUriIgnoreQueryOrderErrorFormat, err)
-				newUri = c.Request.URI().String()
-			}
+// NewCacheByRequestURIWithIgnoreQueryOrder a shortcut function for caching response by uri and ignore query param order.
+func NewCacheByRequestURIWithIgnoreQueryOrder(defaultCacheStore persist.CacheStore, defaultExpire time.Duration, opts ...Option) app.HandlerFunc {
+	cacheStrategy := func(ctx context.Context, c *app.RequestContext) (bool, Strategy) {
+		newUri, err := getRequestUriIgnoreQueryOrder(string(c.Request.RequestURI()))
```
Comment (welkeyever) [inline cache.go:207]: Why use `.RequestURI()` instead of the `.URI()`?

Comment (welkeyever) [reply cache.go:207]: Do not need Scheme and Host anymore?

**Comment by welkeyever** [comment]
Take a look at staticcheck.

File: cache.go (hunk 1 above)
Comment (li-jin-gou) [reply cache.go:207]: Yes, aligned with Gin, Gin's URI is `/a/b?a=1`, but hertz is not. Many migration users assume that `c.Request.URI().String()` and gin behave the same way, but they don't.

**Comment by li-jin-gou** [comment]
PATL @chaoranz758

**Comment by chaoranz758** [comment]
> PATL @chaoranz758

LGTM~

**Review by chaoranz758 — APPROVED** [review]

**Pull request merged**